src/containers/Home/components/ProductForm/index.jsx

```diff
@@ -0,0 +1,178 @@
+      dispatch({ type: 'UPDATE_PRODUCT', payload: formData });
+      toast.success('Product updated successfully!');
+      setFormData({
+        title: '',
+        price: '',
+        image: null,
+        description: '',
+      });
+      setFetched(false);
+    } else {
+      const newProduct = {
+        title: formData.title,
+        price: parseFloat(formData.price),
+        description: formData.description,
+        image: formData.image,
+        id: Math.random().toString(36).substr(2, 9),
+      };
+
+      toast.success('Product added successfully!');
+
+      document.getElementsByName('title')[0].classList.remove('is-valid');
+      document.getElementsByName('price')[0].classList.remove('is-valid');
+      document.getElementById('desc').classList.remove('is-valid');
+      document.getElementsByName('image')[0].classList.remove('is-valid');
+      document.getElementsByName('image')[0].value = null;
+
+      dispatch({ type: 'ADD_PRODUCT', payload: newProduct });
+    }
+    setFormData({
+      title: '',
+      price: '',
+      image: null,
+      description: '',
+    });
+  };
+
+  return (
+    <div>
+      <div className="text-center">
+        <button
+          className="btn-primary btn m-5"
+          onClick={(e) => setShowAddForm(!showAddForm)}
+        >
+          Add Product
+        </button>
+        <button
+          className="btn-primary btn m-5"
+          onClick={(e) => setShowEditForm(!showEditForm)}
+        >
+          Edit Product
+        </button>
+      </div>
+      {showAddForm && (
+        <div className="row justify-content-center">
+          <Form
+            handleSubmit={handleSubmit}
+            handleChange={handleChange}
+            handleImageChange={handleImageChange}
+            formData={formData}
+          />
+        </div>
+      )}
+      {showEditForm && (
+        <div>
+          <div className="text-center">
+            <input
+              name="fetch"
+              id="fetch"
+              className="text-center p-1"
+              type="text"
+              placeholder="Enter ID"
+            />
+            <button
+              className="btn btn-success ml-4"
+              onClick={(e) => {
+                let product = products.filter(
+                  (product) => 
+                    String(product.id) === String(document.getElementById('fetch').value)
+                  )[0];
+                if (product === undefined) {
+                  toast.error('Product not found.');
+                } else {
+                  setFetched(true);
+                  setFormData(product);
+                }
+              }}
+            >
+              Fetch
+            </button>
+          </div>
+          {fetched && (
+            <div className="row justify-content-center mt-4">
+              <Form
+                handleSubmit={handleSubmit}
+                handleChange={handleChange}
+                formData={formData}
+              />
```

Comment:
We can use single Form component for edit & add as we are passing  & handling same data for both form's field